ckanext/dcat/profiles.py

```diff
@@ -307,6 +308,35 @@ def _spatial(self, subject, predicate):
             'geom': geom,
         }
 
+    def _license(self, dataset_ref):
+        '''
+        Returns a license identifier if one of the distributions license is
+        found in CKAN license registry. If no distribution's license matches,
+        None is returned.
+
+        The first distribution with a license found in the registry is used so
+        that if distributions have different licenses we'll only get the first
+        one.
+        '''
+        license_uri2id = {}
+        license_title2id = {}
+        for license_id, license in LicenseRegister().items():
```

Comment:
This only needs to be run once right? There's no need to run the same loop on every dataset (and there could be potentially loads of them when harvesting). So you could do something like:

```python 
if not self._license_uri2id:
    for license_id, license in LicenseRegister().items():
         self._license_uri2id[license.url] = license_id
         # ...
```